### PR TITLE
[Serialization] Add remove_duplicate functions

### PR DESCRIPTION
**Pull request by jmirabel**

### CODE REVIEW

File: include/hpp/pinocchio/serialization.hh [hunk 1]
```diff
@@ -89,4 +93,187 @@ inline void serialize(Archive & ar, Eigen::Array<_Scalar,_Rows,_Cols,_Options,_M
...
+
+template<typename Derived>
+struct eigen_compare {
+  bool operator() (const Eigen::DenseBase<Derived>& a,
+                   const Eigen::DenseBase<Derived>& b)
+  {
+    if (a.size() < b.size()) return true;
+    if (a.size() > b.size()) return false;
+    for(Eigen::Index i = 0; i < a.size(); ++i) {
+      if (a.derived().data()[i] < b.derived().data()[i]) return true;
```
Comment (jcarpent) [inline include/hpp/pinocchio/serialization.hh:112]: It should be PlainObjectBase if you refer to data ;)

**Comment by florent-lamiraux** [comment]
I have a compilation error in humanoid-robot.cc I fixed it by adding
#include <hpp/util/serialization.hh> in humanoid-robot.hh.

**Comment by jmirabel** [comment]
Thanks for your feedback. I'll clean my install and address your comments on Monday.

File: include/hpp/pinocchio/serialization.hh (hunk 1 above)
Comment (jmirabel) [reply include/hpp/pinocchio/serialization.hh:108]: Done. Thanks.

**Pull request merged**